addons/hr_contract/wizard/hr_departure_wizard.py

```diff
@@ -8,29 +8,45 @@
 class HrDepartureWizard(models.TransientModel):
     _inherit = 'hr.departure.wizard'
 
-    def _get_employee_departure_date(self):
-        employee = self.env['hr.employee'].browse(self.env.context['active_id'])
+    def _get_employee_departure_date(self, employee):
         if employee.contract_id.state == "open":
             return False
         expired_contract = self.env['hr.contract'].search([('employee_id', '=', employee.id), ('state', '=', 'close')], limit=1, order='date_end desc')
         if expired_contract:
             return expired_contract.date_end
-        return super()._get_employee_departure_date()
+        return super()._get_employee_departure_date(employee)
 
     set_date_end = fields.Boolean(string="Set Contract End Date", default=lambda self: self.env.user.has_group('hr_contract.group_hr_contract_manager'),
         help="Set the end date on the current contract.")
 
     def action_register_departure(self):
         """If set_date_end is checked, set the departure date as the end date to current running contract,
         and cancel all draft contracts"""
-        current_contract = self.sudo().employee_id.contract_id
-        if current_contract and current_contract.date_start > self.departure_date:
+        employees = self.employee_ids
+        Contract = self.env['hr.contract'].sudo()
```

Comment:
be careful with that placing a sudo without explicitly naming it that way could lead to misuse of it